processing/src/main/java/org/apache/carbondata/processing/store/TablePage.java

```diff
@@ -104,19 +105,27 @@
         page.setStatsCollector(KeyPageStatsCollector.newInstance(DataTypes.BYTE_ARRAY));
         dictDimensionPages[tmpNumDictDimIdx++] = page;
       } else {
+        // will be encoded using string page
+        LocalDictionaryGenerator localDictionaryGenerator =
+            model.getColumnLocalDictGenMap().get(spec.getFieldName());
         if (DataTypes.VARCHAR == spec.getSchemaDataType()) {
-          page = ColumnPage.newPage(spec, DataTypes.VARCHAR, pageSize);
+          page = ColumnPage.newLocalDictPage(spec,
```

Comment:
if the `localDictionaryGenerator` is null, for VARCHAR also, it should not generate localDictPage, null check is missing i think